File: pset1/cash/C/cash.c
#include <stdio.h>
#include <math.h>
int main(void)
{
    int coins = 0;
    //prompts user for change owed as x and makes sure it is positive
    float x;
    do
    {
        printf("Change Owed: ");
        scanf("%f", &x);
    }
    while(x < 0.00);

    //amounts x to cents by multiplying by 100
    x = round(x * 100);
    //quarters
    while(x >= 25)
    {
        coins += 1;
        x -= 25;
    }
    //dimes
    while(x >= 10)
    {
        coins += 1;
        x -= 10;
    }
    //nickels
    while(x >= 5)
    {
        coins += 1;
        x -= 5;
    }
    //pennies
    while(x >= 1)
    {
        coins += 1;
        x -= 1;
    }
    //prints minimum number of coins given back
    printf("%d\n", coins);
    return 0;
}

File: pset1/cash/python/cash.py
# Get change owed
while True:
    x = float(input("Change Owed: "))
    if x == 0:
        print(0)
    elif x < 0:
        continue
    else:
        break
# Convert x to cents
change = round(x * 100)
coins = 0

# Quarters
while change >= 25:
    coins += 1
    change -= 25

# Dimes
while change >= 10:
    coins += 1
    change -= 10

# Nickles
while change >= 5:
    coins += 1
    change -= 5

# Pennies
while change >= 1:
    coins += 1
    change -= 1

print(coins)